frontend/src/metabase/dashboard/components/AutoApplyFiltersToast/AutoApplyFiltersToast.tsx

```diff
@@ -74,8 +79,22 @@ export default function AutoApplyFilterToast() {
           ),
         }),
       );
+
+      return () => {
+        dispatch(dismissAutoApplyFiltersToast());
+        dispatch(dismissUndo(autoApplyFiltersToastId, false));
+      };
     }
-  }, [autoApplyFiltersToastStateName, dashboardId, dispatch]);
+  }, [
+    autoApplyFiltersToastId,
+    autoApplyFiltersToastStateName,
+    dashboardId,
+    dispatch,
+  ]);
+
+  useUnmount(() => {
+    dispatch(setNeverShowAutoApplyFiltersToast());
```

Comment:
We should also reset the auto-apply filters toast state name on unmount, otherwise, if we revisit the dashboard after navigating away from the dashboard, the auto-apply filters toast state could be in a weird state. e.g. when revisiting the dashboard page and instantly getting the auto-apply filters toast since we hadn't reset the state name.